### PR TITLE
Add types for cronitor

### PR DESCRIPTION
**Pull request by alexpresthus**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## cronitor ([<kbd>unpkg</kbd>](https://unpkg.com/browse/cronitor@latest/))
was missing the following properties:
1. Cronitor

## Formatting

The following files are not formatted:
1. types/cronitor/index.d.ts
1. types/cronitor/cronitor-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c859a697da7225eee31721280ef4c1ca96575d42
</p>

File: types/cronitor/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        FAIL: "fail";
+        OK: "ok";
+    };
+}
+
+declare function Cronitor(apiKey: string, config?: CronitorConfig): Cronitor;
+
+export = Cronitor;
+
+export as namespace Cronitor;
```
Comment (sheetalkamat) [inline types/cronitor/index.d.ts:115]:
```suggestion
```

It doesnt look like from documentation that the library supports using it as global scope. 
If it does please point to documentation.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/cronitor/index.d.ts (hunk 1 above)
Comment (alexpresthus) [reply types/cronitor/index.d.ts]: You're absolutely right. Fixed!

File: types/cronitor/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
+type MonitorState = "run" | "complete" | "fail" | "ok";
```
Comment (MysteryBlokHed) [inline types/cronitor/index.d.ts:1]: If you want these types to be available to consumers, you can wrap everything in a `declare namespace Cronitor` block (the namespace name should be the same as the thing being exported with `export =`).

The only other change that would need to be made would be changing line 111 to reference `Cronitor.CronitorConfig` instead

File: types/cronitor/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    Monitor: typeof Monitor;
+    Event: typeof Event;
+
+    static State: {
+        RUN: "run";
+        COMPLETE: "complete";
+        FAIL: "fail";
+        OK: "ok";
+    };
```
Comment (sheetalkamat) [inline types/cronitor/index.d.ts:108]:
```suggestion
```

There is no member called State

``` js
//a.js
const x = require("cronitor");
console.log(x.State)
```


```
c:\temp\test>node a.js
undefined
```

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/cronitor/index.d.ts (hunk 3 above)
Comment (alexpresthus) [reply types/cronitor/index.d.ts]: You're right, the State() getter is only on Cronitor.Monitor. Fixed in latest commit.

File: types/cronitor/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            body?: string;
+            headers?: Record<string, string>;
+            cookies?: Record<string, string>;
+            timeout_seconds?: number;
+            follow_redirects?: boolean;
+            verify_ssl?: boolean;
+        }; // required by type: check
+    }
+
+    class Monitor {
```
Comment (sheetalkamat) [inline types/cronitor/index.d.ts:70]: `Monitor` and `Event` cannot be class exposed here as they are not exported by the library. You should move them outside the namespace.

```js
const x = require("cronitor");
console.log(x. Monitor )
console.log(x. Event )
```

```
> c:\temp\test>node a.js
undefined
undefined
```

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Review by jakebailey — CHANGES_REQUESTED** [review]
After #67332, this PR will need to be updated to include an `.npmignore`. See https://github.com/DefinitelyTyped/DefinitelyTyped#npmignore

File: types/cronitor/index.d.ts (hunk 4 above)
Comment (alexpresthus) [reply types/cronitor/index.d.ts]: Right - this is fixed now

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/cronitor/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6088b1234efb58591ead278763368bed7f3013e2
</p>

**Review by gabritto — APPROVED** [review]

**Comment by alexpresthus** [comment]
Ready to merge

**Pull request merged**